Dockerfile

```
@@ -79,37 +76,49 @@ RUN conan export conan/mqtt_cpp \
 FROM os AS release
 
 LABEL author="mtconnect" description="Docker image for the latest Production MTConnect C++ Agent"
+ARG BIN_DIR=/usr/bin
+ARG MTCONNECT_DIR=/etc/mtconnect
+
+ENV MTCONNECT_DIR=$MTCONNECT_DIR
 
 # install ruby for simulator
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y ruby zip
 
 # change to a new non-root user for better security.
 # this also adds the user to a group with the same name.
 # --create-home creates a home folder, ie /home/<username>
 RUN useradd --create-home agent
 USER agent
+WORKDIR /home/agent
 
 # install agent executable
-COPY --chown=agent:agent --from=build /root/agent/build/bin/agent /usr/local/bin/
+COPY --chown=agent:agent --from=build /root/agent/*.zip .
 
 # copy data to /etc/mtconnect
-COPY --chown=agent:agent --from=build /root/agent/schemas /etc/mtconnect/schemas
-COPY --chown=agent:agent --from=build /root/agent/simulator /etc/mtconnect/simulator
-COPY --chown=agent:agent --from=build /root/agent/styles /etc/mtconnect/styles
+RUN unzip *.zip
+
+ENV ZIP_DIR=/home/agent/agent-*-Linux
+
+USER root
+RUN cp $ZIP_DIR/bin/* $BIN_DIR
+RUN mkdir -p $MTCONNECT_DIR && chown -R agent:agent $MTCONNECT_DIR
+
+USER agent
+RUN cp -r $ZIP_DIR/schemas $ZIP_DIR/simulator $ZIP_DIR/styles $ZIP_DIR/demo $MTCONNECT_DIR/
```

In BASH, we should always quote variables (we should do this in the entire `Dockerfile`). And we could also use BASH globbing here.

```suggestion
RUN cp -r "$ZIP_DIR"/{demo,schemas,simulator,styles} "$MTCONNECT_DIR"
```

---

Oh, forget about this, it won’t work, as Docker build usines `/bin/sh` instead of `/bin/bash` as its shell. :man_facepalming: